File: 11.10/이아름/멀쩡한 사각형.cpp
#include <algorithm>
using namespace std;

int GCD(int a, int b) {
	if (b < 1) {
		return a;
	}
	return GCD(b, a%b);
}

long long solution(int w, int h) {
	long long answer = (long long)h*w;
	int gcd = GCD(max(w, h), min(w, h));
	int minus = (h + w - gcd);
	return answer - (long long)minus;
}

File: 11.10/정호조/멀쩡한사각형.java
import java.util.*;

public class 멀쩡한사각형 {

	public static void main(String[] args) {
		System.out.println(solution(8, 12));
	}

	public static long solution(int w, int h) {
		long answer = 1;

		long tmp = Math.max(w, h);
		long tmp2 = Math.min(w, h);

		w = (int) tmp2;
		h = (int) tmp;
		answer = w * h - (w + h - getGCD(h, w));

		return answer;
	}

	static int getGCD(int x, int y) {
		if (x % y == 0)
			return y;

		return getGCD(y, x % y);
	}
}
